compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -758,7 +759,17 @@ setReductionVectorDistributionConfig(IREE::GPU::TargetAttr target,
 
   SmallVector<int64_t> bounds = op.getStaticLoopRanges();
   IREE::GPU::TargetWgpAttr wgp = target.getWgp();
-  int64_t reductionSize = bounds[reductionDims.back()];
+  // Consider the entire reduction dimension. For example, there are multiple
+  // reductions in scaled matmul with the last dimension being the block size
+  // (32 for gfx950).
+  int64_t reductionSize = 1;
+  for (unsigned dim : reductionDims) {
+    if (ShapedType::isDynamic(bounds[dim])) {
+      reductionSize = ShapedType::kDynamic;
+      break;
+    }
+    reductionSize *= bounds[dim];
+  }
```

Comment:
This seems correct, but it looks like some logic needs to be updated somewhere else too.